src/Core/Analyser/DependencyLayersAnalyser.php

```diff
@@ -35,14 +35,14 @@ public function __construct(
     /**
      * @throws AnalyserException
      */
-    public function process(): Result
+    public function process(): AnalysisResultBuilder
```

Comment:
This part feels a bit off to me in combination with the usage in the `AnalyseRunner`. I think this can either be resolved by renaming `AnalysisResultBuilder` and its `build()` method to something like `AnalysisResult` and `toOutputResult()` or moving the `build()` in here and not [have the chained call in the `AnalyseRunner`](https://github.com/qossmic/deptrac/pull/1091/files#diff-5735c5b65057ba230dbf56531a1534e35e25e825e2aa89d757f2dca86e119982R50)

I think I would prefer the formaer and move the `build()/toOutputResult()` to where its needed, i.e. when we pass the result to the output formatter.